genconfig.py

```diff
@@ -1,42 +1,60 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import argparse
 import os
 import yaml
-
-scylla_port = 9180
-node_exporter_port = 9100
-
-
-def append_port(ips, port):
-    return ["%s:%s" % (x, port) for x in ips]
-
-
-def gen_targets(servers, port):
-    return {"targets": append_port(servers, port)}
-
-
-def dump_yaml(directory, filename, servers, port):
+import re
+
+def gen_targets(servers, cluster):
+    if ':' not in servers:
+        raise Exception('Server list must contain a dc name')
+    dcs = servers.split(':')
+    res = {"labels": {"cluster": cluster, "dc": dcs[0]}}
+    res["targets"] = dcs[1].split(',')
+    return res;
+
+def get_servers_from_nodetool_status(filename):
+    res = []
+    dc = None
+    ips = []
+    with open(filename, 'r') as status_file:
+        for l in status_file.readlines():
+            if dc:
+                ip = re.search(r"..  ([\d\.]+)\s", l)
+                if ip:
+                    ips.append(ip.group(1))
+            m = re.search(r"Datacenter: (.*)$", l)
+            if m:
+                if dc:
+                    res.append(dc + ":" + ",".join(ips))
+                ips = []
+                dc = m.group(1)
+    if dc:
+        res.append(dc + ":" + ",".join(ips))
+    return res
+
+
+def dump_yaml(directory, filename, servers, cluster):
     try:
         os.mkdir(directory)
-    except OSError, err:
+    except OSError as err:
         if err.errno != 17:
             raise
         pass
-    stream = file(os.path.join(directory, filename), 'w')
-    yaml.dump([gen_targets(servers, port)], stream)
+    with open(os.path.join(directory, filename), 'w') as yml_file:
+        yaml.dump([gen_targets(s, cluster) for s in servers], yml_file, default_flow_style=False)
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description="Generate configuration for prometheus")
     parser.add_argument('-d', '--directory', help="directory where to generate the configuration files", type=str, default="./")
-    parser.add_argument('-s', '--scylla', help="Generate scylla_servers.yml file", action='store_true')
-    parser.add_argument('-n', '--node', help="Generate node_exporter_servers.yml file", action='store_true')
-    parser.add_argument('servers', help="list of nodes to configure, separated by space", nargs='+', type=str, metavar='node_ip')
+    parser.add_argument('-c', '--cluster', help="The cluster name", type=str, default="my-cluster")
```

Comment:
where is the cluster name used?

Reply:
It's not a must (that's why I'm adding one as default) but it can be used when monitoring multiple clusters